src/components/policy/policy_external/src/cache_manager.cc

```diff
@@ -705,6 +708,7 @@ bool CacheManager::ApplyUpdate(const policy_table::Table& update_pt) {
   pt_->policy_table.consumer_friendly_messages.assign_if_valid(
       update_pt.policy_table.consumer_friendly_messages);
 
+  UnwrapAppPolicies(pt_->policy_table.app_policies_section.apps);
```

Comment:
@AKalinich-Luxoft 
I didn't catch why you call this method again in several places. AFAIK the purpose of this method is to replace permissions records in PTU (not in current PT) like
```
"app_id" : "default"
```
with the content of "default" permissions section.
So the only intention was to call it once PTU is received and before processing updated permissions i.e. current PT should not be 'unwrapped' again.

Reply:
@dev-gh as I found out current PT should be re-unwrapped again after PTU applying. Let's imagine such case:
1. SDL is started with preloaded pt
2. App1 is registered and gets "default" policy permissions. Note that ApplicationPolicies class wrapped in Stringifyable class and its predefined_string_ is also set to "default"
3. App1 initiates PTU where only default policy permissions were changed (for example was added new functional group)
4. As you said, before processing updated permissions, policy manager calls UnwrapAppPolicies. This function replaces application policies with default policies for applications which have "default" predefined_string string (see step 2)
5. Before applying changes in PT we have "App1" with copied default permissions and "default" with its own permissions.
6. All PTU changes was applied. Now "default" has updated permissions, however "App1" still have copy of default permissions before update, i.e. default permission changes will not affect App1. This changes will take effect for App1 only after SDL restart. 
7. I think this is not correct behaviour so I add UnwrapAppPolicies call after updates processing, so this call re-assigns updated default policy permissions for specific applications with default permissions, so with this rewrapping it works as expected and updated permissions will take an effect immediately for App1.

This is a reason why I use this function here. I will be thankful if you could suggest better solution here.

Reply:
@AKalinich-Luxoft 
I think p.6 is incorrect, I mean "Now "default" has updated permissions, however "App1" still have copy of default permissions before update, i.e. default permission changes will not affect App1."
If it is so, then there is some other issue with applying of PTU.

Per my understanding flow must be following:
- pp. 1-3 are okay
 - p. 4 - clarification: replacement is done for PTU, not for current PT, so during checking of permissions difference App1 should have old 'default' permissions in PT but new 'default' in PTU 
 - p. 5 - okay
 - p. 6 - during applying of PTU App1 policy must be replaced with the one from PTU i.e.  groups must be updated as well so I don't quite understand why restart is required

From the code in 'develop' branch I see that in case new groups detected and no user consents are required RESULT_CONSENT_NOT_REQIURED code is returned during difference check. Looks like this code doesn't trigger OnPermissionsChange sending and I think it may be related to issue cause.

Reply:
@dev-gh p.4 - what if we send PTU, where section with App1 will not be present? How to check permission changes for App1 in this case?
regarding p.6 - it is okay only if App1 is present at least in PTU, but what if it will not be present there as in case above?

Reply:
@dev-gh restart of SDL is required because in local PT permissions will not be updated for App1 with default policy after PTU. However if SDL will be restarted, it will load local PT from storage, and when we register App1 again, it will have updated default permissions as expected.

Reply:
@AKalinich-Luxoft 
If SDL sends snapshot to backend with some application id then expected to get back specific permissions or "default". Otherwise it is wrong PTU.

Reply:
@dev-gh ok, in this case I will check this issue with PTU with "default" for some application id. Probably some of this logic is redundant in this case.

Reply:
@dev-gh pull request was updated and redundant logic were removed. Please review again 89ceac249b54ec02c042865b27f4af8b2363d450